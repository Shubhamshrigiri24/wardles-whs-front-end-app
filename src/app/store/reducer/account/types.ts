import { TActionStatus } from "app/store/types";
import { ACCOUNT_ACTIONS } from "app/store/reducer/account/constants";

export type TAccount = {
  dob: string;
  email: string;
  firstName: string;
  gender: string;
  lastName: string;
  phone: string;
  updateTime: string;
  userId: string;
  addressline1: string;
  addressline2: string;
  agreeMarketing: boolean;
  city: string;
  gpDetails: string;
  postcode: string;
};

export type State = {
  data: TAccount | null;
  status: TActionStatus;
};

export type SetAccountDataActionType = {
  type: typeof ACCOUNT_ACTIONS.SET_ACCOUNT;
  payload: TAccount;
};

export type SetAccountStatusActionType = {
  type: typeof ACCOUNT_ACTIONS.SET_ACCOUNT_STATUS;
  payload: TActionStatus;
};

export type AccountActionTypes =
  | SetAccountDataActionType
  | SetAccountStatusActionType;
