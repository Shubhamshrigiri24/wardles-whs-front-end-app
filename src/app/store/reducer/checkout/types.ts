import { RSAA } from "redux-api-middleware";
import { PostcodeLookupAddress } from "../shared";

export interface State {
  customerDetails: CustomerDetails;
  confirmedPayment: boolean;
  braintreeToken: string;
  paymentErrorMessage: string;
  postCodeLookUpAddresses: PostcodeLookupAddress[];
  deliveryDetails: DeliveryDetails;
}

export interface CustomerDetails {
  title?: string;
  firstName: string;
  lastName: string;
  line1: string;
  line2: string;
  city: string;
  postcode: string;
  email: string;
  phone: string;
  surgeryName?: string;
  surgeryAddress?: string;
  allowMarketing?: boolean;
}

export interface AddCustomerDetailsAction {
  type: string;
  payload: {
    customerDetails: CustomerDetails;
  };
}

export interface DeliveryDetails {
  title?: string;
  firstName: string;
  lastName: string;
  line1: string;
  line2: string;
  city: string;
  postcode: string;
}

export interface AddDeliveryDetailsAction {
  type: string;
  payload: {
    deliveryDetails: DeliveryDetails;
  };
}

export interface FetchBraintreeTokenAction {
  type: string;
  [RSAA]: {
    endpoint: string;
    method: string;
    types: [string, string, string];
  };
}

export interface FetchBraintreeTokenResponseAction {
  type: string;
  payload: {
    token: string;
  };
}

export interface PaymentFailureAction {
  type: string;
  payload: {
    response: {
      message: string | null;
    } | null;
  };
}

export interface FetchAddressFromPostcodeActionResponse {
  type: string;
  payload: {
    addresses: PostcodeLookupAddress[];
  };
}
