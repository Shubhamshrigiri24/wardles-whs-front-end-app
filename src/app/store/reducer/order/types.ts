import { Pack as OnlinePack, Product as OnlineProduct } from "../online/types";
import { ORDER_ACTIONS } from "app/store/reducer/order/constants";

// Easy enough passthrough for props. When needed, we can break this off.

export type Pack = OnlinePack;

export type PackDiscount = {
  code: string;
  type: "percentage" | "fixed";
  discountInfo: number;
  discountAmount: number;
  priceWithDiscount: number;
};

export type Discount = {
  initial: number;
  discountValue: number;
  final: number;
  discounts?: null | Array<{
    code: string;
    type: PackDiscount["type"];
    amount: number;
    calculated: {
      initial: number;
      value: number;
      final: number;
    };
  }>;
};

export type Product = OnlineProduct & {
  subscription?: boolean;
  image?: string;
  delivery?: string;
  orderId?: string;
  subscriptionId?: string;
};

export type OrderCustomer = {
  customerType: string;
  customerDetails: null | OrderCustomerDetails;
};

export type OrderProduct = Product | null;

export interface OrderCustomerDetails {
  firstName: string;
  lastName: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  postcode: string;
  email: string;
  phone: string;
  dob?: string;
  gpDetails?: string;
  agreeMarketing: boolean;
}

export type State = {
  customer: OrderCustomer;
  product: OrderProduct;
};

export type TOrder = State;

export type SetCustomerActionType = {
  type: typeof ORDER_ACTIONS.SET_CUSTOMER;
  payload: OrderCustomer;
};

export type SetProductActionType = {
  type: typeof ORDER_ACTIONS.SET_PRODUCT;
  payload: OrderProduct;
};

export type CleanupOrderFlowStateActionType = {
  type: typeof ORDER_ACTIONS.CLEANUP_ORDER_FLOW_STATE;
};

export type OrderActionTypes =
  | SetCustomerActionType
  | SetProductActionType
  | CleanupOrderFlowStateActionType;
