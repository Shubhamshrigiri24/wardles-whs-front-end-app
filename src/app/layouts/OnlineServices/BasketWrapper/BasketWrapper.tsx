import React, { useState } from "react";
import {
  Container,
  Divider,
  IconButton,
  makeStyles,
  Theme,
  useTheme,
  useMediaQuery,
  Typography,
} from "@material-ui/core";
import ShoppingCart from "@material-ui/icons/ShoppingCart";
import CloseIcon from "@material-ui/icons/Close";
import Basket from "./Basket/Basket";
import SplitPanel from "../../../../components/SplitPanel/SplitPanel";
import {
  Product,
  Pack,
  OnlineService,
} from "../../../store/reducer/online/types";
import WellLogo from "./Well.svg";
import { useRouteMatch } from "react-router-dom";
import OnlineFooter from "../../../../components/OnlineFooter/OnlineFooter";

const useStyles = makeStyles((theme: Theme) => ({
  container: {
    padding: theme.breakpoints.down("sm") ? theme.spacing(1) : theme.spacing(6),
  },
  header: {
    display: "flex",
    marginLeft: theme.spacing(19),
    alignItems: "center",
    height: theme.spacing(14),
    [theme.breakpoints.down("sm")]: {
      height: theme.spacing(8),
      justifyContent: "space-between",
      margin: theme.spacing(1.5, 2.5),
    },
    color: "#65727D",
    fontSize: "18px",
    lineHeight: "32.4px",
  },
  headerConfirmation: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    marginLeft: theme.spacing(19),
    height: theme.spacing(14),
    [theme.breakpoints.down("sm")]: {
      height: theme.spacing(8),
      margin: theme.spacing(1.5, 2.5),
    },
    color: "#65727D",
    fontSize: "18px",
    lineHeight: "32.4px",
  },
  divider: {
    width: "100%",
  },
  flex: {
    display: "flex",
    justifyContent: "space-between",
  },
  logoContainer: {
    display: "flex",
    flexDirection: "column",
  },
  wellLogo: {
    maxHeight: "35px",
    maxWidth: "90px",
    objectFit: "contain",
    paddingBottom: theme.spacing(1),
  },
  textContainer: {
    marginLeft: theme.spacing(2),
  },
  overlay: {
    width: "100%",
    height: "100vh",
    backgroundColor: "rgba(0, 0, 0, 0.5)",
  },
  floatingBasket: {
    backgroundColor: "white",
  },
  mobileBasketWrapper: {
    position: "absolute",
    width: "100%",
    zIndex: 100,
    height: "100vh",
  },
  showBasketButton: {
    color: "#0C161F",
    display: "flex",
    alignItems: "center",
  },
  basketCount: {
    marginLeft: theme.spacing(1),
  },
}));

export interface BasketWrapperStateProps {
  selectedOnlineService: OnlineService;
  product: Product;
  basketItem: Pack;
  children: Element[];
  hasFooter?: boolean;
}

const BasketWrapper: React.FC<BasketWrapperStateProps> = ({
  selectedOnlineService,
  product: getProduct,
  basketItem: getBasketItem,
  children,
  hasFooter,
}) => {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  const [showBasket, setShowBasket] = useState(false);
  const classes = useStyles();
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
  // eslint-disable-next-line @typescript-eslint/naming-convention
  const showMobileBasket = isMobile && showBasket;
  const isOnlineConfirmationRoute = useRouteMatch({
    path: "/:onlineServiceId/checkout/confirmation",
    strict: true,
    sensitive: true,
  });

  const toggleBasket = () => {
    document.body.style.overflowY = showBasket ? "visible" : "hidden";
    setShowBasket(!showBasket);
  };

  // Once we have multiple items in a basket we can change this
  const basketItems = [getBasketItem];

  return isOnlineConfirmationRoute ? (
    <div>
      <div className={classes.headerConfirmation}>
        <div className={classes.logoContainer}>
          <img src={WellLogo} alt={"WellLogo"} className={classes.wellLogo} />
          <div className={classes.textContainer}>
            <Typography>Pharmacy Services</Typography>
          </div>
        </div>
      </div>
      <Divider className={classes.divider} />
      {children}
    </div>
  ) : (
    <>
      <div className={classes.header}>
        <div className={classes.logoContainer}>
          <img src={WellLogo} alt={"WellLogo"} className={classes.wellLogo} />
          <div className={classes.textContainer}>
            <Typography>Pharmacy Services</Typography>
          </div>
        </div>
        {isMobile && (
          <div>
            <IconButton data-testid={"close"} onClick={toggleBasket}>
              {showBasket && <CloseIcon height={"21px"} />}
              {!showBasket && (
                <div className={classes.showBasketButton}>
                  <ShoppingCart height={"21px"} />
                  <Typography className={classes.basketCount}>
                    {basketItems.length}
                  </Typography>
                </div>
              )}
            </IconButton>
          </div>
        )}
      </div>
      <Divider className={classes.divider} />
      <div>
        {showMobileBasket && (
          <div className={classes.mobileBasketWrapper}>
            <div className={classes.floatingBasket}>
              <Basket
                product={getProduct}
                // TODO update to reflect a new reducer that shows multiple items in the basket.
                basketItems={basketItems}
                onlineService={selectedOnlineService}
              />
            </div>
            <div className={classes.overlay} onClick={toggleBasket} />
          </div>
        )}
        <SplitPanel
          main={
            <Container
              className={classes.container}
              maxWidth={"md"}
              disableGutters
            >
              {children}
            </Container>
          }
          secondary={
            <>
              {!isMobile && (
                <Basket
                  product={getProduct}
                  // TODO update to reflect a new reducer that shows multiple items in the basket.
                  basketItems={basketItems}
                  onlineService={selectedOnlineService}
                />
              )}
            </>
          }
        />
      </div>
      {hasFooter && <OnlineFooter />}
    </>
  );
};

export default BasketWrapper;
