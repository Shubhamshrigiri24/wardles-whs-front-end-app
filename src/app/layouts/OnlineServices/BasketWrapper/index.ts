import { connect } from "react-redux";
import { RootState } from "app/store/types";
import BasketWrapper, { BasketWrapperStateProps } from "./BasketWrapper";
import {
  getSelectedPack,
  getSelectedProduct,
  getSelectedOnlineService,
} from "../../../store/selectors";

function mapStateToProps(
  state: RootState,
  ownProps: any
): BasketWrapperStateProps {
  return {
    selectedOnlineService: getSelectedOnlineService(state),
    product: getSelectedProduct(state),
    basketItem: getSelectedPack(state),
    ...ownProps,
  };
}

export default connect(mapStateToProps)(BasketWrapper);
