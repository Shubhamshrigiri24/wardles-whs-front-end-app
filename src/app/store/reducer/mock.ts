import faker from "faker/locale/en_GB";
import { CustomerDetails, DeliveryDetails } from "./checkout/types";
import { PostcodeLookupAddress } from "./shared";
import {
  OnlineQuestion,
  Pack,
  Product,
  SuccessfulConsultationResponse,
} from "./online/types";

export function makeCustomerDetails(
  override?: Partial<CustomerDetails>
): CustomerDetails {
  return {
    firstName: "firstName",
    lastName: "lastName",
    email: "a@b.com",
    postcode: "M19 2UL",
    line1: "line1",
    line2: "line2",
    city: "city",
    phone: "01189998819991197253",
    surgeryName: "surgeryName",
    surgeryAddress: "surgeryAddress",
    allowMarketing: true,
    ...override,
  };
}

export function makeDeliveryDetails(): DeliveryDetails {
  return {
    title: "Mrs",
    firstName: faker.name.firstName(),
    lastName: faker.name.lastName(),
    postcode: faker.address.zipCode(),
    line1: faker.address.streetName(),
    line2: faker.address.streetName(),
    city: faker.address.city(),
  };
}

export function makePostCodeLookUpAddresses(): PostcodeLookupAddress[] {
  return [
    {
      line1: "64 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
    {
      line1: "66 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
    {
      line1: "68 Broom Avenue",
      line2: "",
      city: "Manchester",
      country: "England",
      postcode: "M19 2UL",
    },
  ];
}

export function makeSuccessfulConsultationResponse(
  id: string = "default",
  override?: Partial<SuccessfulConsultationResponse>
): SuccessfulConsultationResponse {
  return {
    answer: `answer-${id}`,
    answerString: `answerString-${id}`,
    initialQuestion: `question-${id}`,
    kind: `trepetto-input-type-${id}`,
    question: `question-${id}`,
    type: "string",
    ...override,
  };
}

export function makeOnlineQuestion(number: number = 1): OnlineQuestion {
  return {
    number,
    question: `question-${number}`,
    answer: `answer-${number}`,
    type: "string",
  };
}

export function makeOnlineQuestions(): OnlineQuestion[] {
  return [makeOnlineQuestion(1), makeOnlineQuestion(2), makeOnlineQuestion(3)];
}

export function makeProduct(
  id: string = "default",
  override?: {
    id?: string;
    name?: string;
    service?: string;
    packs?: Pack[];
    variant?: string;
  }
): Product {
  return {
    id,
    name: `name-${id}`,
    service: `service-${id}`,
    packs: [
      makePack(`${id}-pack-1`),
      makePack(`${id}-pack-2`),
      makePack(`${id}-pack-3`),
    ],
    variant: "",
    ...override,
  };
}

export function makePack(
  id: string = "default",
  override?: {
    sku?: string;
    name?: string;
    label?: string;
    price?: number;
    pricePerUnit?: number;
    itemsPerPack?: number;
  }
): Pack {
  return {
    sku: `sku-${id}`,
    name: `name-${id}`,
    label: `label-${id}`,
    basePrice: 1,
    price: 1,
    pricePerUnit: 1,
    itemsPerPack: 1,
    discounts: [],
    ...override,
  };
}

export const customerDetails: CustomerDetails = {
  firstName: "FIRST_NAME",
  lastName: "LAST_NAME",
  email: "EMAIL",
  postcode: "POSTCODE",
  line1: "LINE_1",
  line2: "LINE_2",
  city: "CITY",
  phone: "PHONE",
  surgeryName: "SURGERY_NAME",
  surgeryAddress: "SURGERY_ADDRESS",
  allowMarketing: true,
};

export const deliveryDetails: DeliveryDetails = {
  title: "Mrs",
  firstName: "FIRST_NAME",
  lastName: "LAST_NAME",
  postcode: "POSTCODE",
  line1: "LINE_1",
  line2: "LINE_2",
  city: "CITY",
};
