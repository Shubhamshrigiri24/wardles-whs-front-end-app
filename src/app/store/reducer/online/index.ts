import { handleActions } from "redux-actions";
import {
  allOnlineServices,
  Ed,
  State,
  Pack,
  Product,
  defaultProduct,
  defaultPack,
  SET_PACK,
  SET_PRODUCT,
  SetPackAction,
  SetProductAction,
  FetchProductsResponseAction,
  FETCH_PRODUCTS_RESPONSE,
  FETCH_PRODUCTS_FAILURE,
  FETCH_PRODUCTS_REQUEST,
  FETCH_BRAINTREE_TOKEN_REQUEST,
  FETCH_BRAINTREE_TOKEN_RESPONSE,
  ON_ORDER_FAILURE,
  ON_ORDER_REQUEST,
  ON_ORDER_RESPONSE,
  ON_ORDER,
  CLEAR_ORDER_FAILURE_ERROR,
  OrderResponseAction,
  SetConsultationAction,
  SetIsConsultationValidForOrderAction,
  OnlineQuestion,
  SetOnlineServiceAction,
  OnlineService,
  SET_ONLINE_SERVICE,
  CHECK_PRODUCT_TOKEN_RESPONSE,
  CHECK_PRODUCT_TOKEN_FAILURE,
  CHECK_PRODUCT_TOKEN_REQUEST,
  CheckProductTokenResponseAction,
  SET_PRODUCT_TOKEN,
  SetProductTokenAction,
  INVALIDATE_PRODUCT_TOKEN,
  SEND_CONSULTATION_REQUEST,
  SEND_CONSULTATION_RESPONSE,
  SEND_CONSULTATION_FAILURE,
  SendConsultationSuccessfulResponse,
} from "./types";
import { combineReducers, Reducer } from "redux";
import { GENERIC_ERROR_MESSAGE } from "app/store/config";
import {
  FetchBraintreeTokenResponseAction,
  PaymentFailureAction,
  SET_CONSULTATION,
  SET_IS_CONSULTATION_VALID_FOR_ORDER,
  START_TRIPETTO,
} from "./types";
import { ORDER_ACTIONS } from "app/store/reducer/order/constants";

const selectedOnlineService = handleActions(
  {
    [SET_ONLINE_SERVICE]: (
      _state: OnlineService,
      action: SetOnlineServiceAction
    ) => {
      return action.payload ? action.payload.selectedOnlineService : Ed;
    },
  },
  Ed
);

const onlineServices = handleActions({}, allOnlineServices);

const selectedPack = handleActions(
  {
    [SET_PACK]: (_state: Pack, action: SetPackAction) => {
      return action.payload ? action.payload.selectedPack : defaultPack;
    },
  },
  defaultPack
);

const selectedProduct = handleActions(
  {
    [SET_PRODUCT]: (_state: Product, action: SetProductAction) => {
      return action.payload ? action.payload.selectedProduct : defaultProduct;
    },
  },
  defaultProduct
);

const products = handleActions(
  {
    [FETCH_PRODUCTS_RESPONSE]: (
      _state: Product[],
      action: FetchProductsResponseAction
    ) => {
      return action.payload ? action.payload : [];
    },
    [FETCH_PRODUCTS_FAILURE]: () => [],
    [FETCH_PRODUCTS_REQUEST]: () => [],
    [START_TRIPETTO]: () => [],
  },
  [defaultProduct]
);

const productToken = handleActions(
  {
    [SET_PRODUCT_TOKEN]: (
      _state: string | null,
      action: SetProductTokenAction
    ) => action.payload,
    [INVALIDATE_PRODUCT_TOKEN]: (_state: string | null) => null,
  },
  null
);

const productTokenValid = handleActions(
  {
    [CHECK_PRODUCT_TOKEN_RESPONSE]: (
      _state: boolean,
      action: CheckProductTokenResponseAction
    ) => {
      return !!action.payload.status;
    },
    [CHECK_PRODUCT_TOKEN_FAILURE]: () => false,
    [CHECK_PRODUCT_TOKEN_REQUEST]: () => true,
  },
  true
);

const fetching = handleActions(
  {
    [FETCH_PRODUCTS_REQUEST]: () => true,
    [FETCH_PRODUCTS_RESPONSE]: () => false,
    [FETCH_PRODUCTS_FAILURE]: () => false,
  },
  true
);

const braintreeToken = handleActions(
  {
    [ON_ORDER]: () => "",
    [FETCH_BRAINTREE_TOKEN_REQUEST]: () => "",
    [FETCH_BRAINTREE_TOKEN_RESPONSE]: (
      _state: string,
      action: FetchBraintreeTokenResponseAction
    ) => action.payload.token,
    [START_TRIPETTO]: () => "",
  },
  ""
);

const orderErrorMessage = handleActions(
  {
    [ON_ORDER_FAILURE]: (_state: string, action: PaymentFailureAction) =>
      action?.payload?.response?.message || GENERIC_ERROR_MESSAGE,
    [FETCH_BRAINTREE_TOKEN_REQUEST]: () => "",
    [ON_ORDER]: () => "",
    [CLEAR_ORDER_FAILURE_ERROR]: () => "",
  },
  ""
);

const successfulOrder = handleActions(
  {
    [ON_ORDER_FAILURE]: () => false,
    [ON_ORDER_REQUEST]: () => false,
    [SET_CONSULTATION]: () => false,
    [ON_ORDER_RESPONSE]: () => true,
    [FETCH_BRAINTREE_TOKEN_REQUEST]: () => false,
  },
  false
);

const consultationResponses = handleActions(
  {
    [SET_CONSULTATION]: (
      _state: OnlineQuestion[],
      action: SetConsultationAction
    ) => action?.payload?.consultation || [],
    [START_TRIPETTO]: () => [],
    [ORDER_ACTIONS.CLEANUP_ORDER_FLOW_STATE]: () => [],
  },
  []
);

const consultationToken = handleActions(
  {
    [SEND_CONSULTATION_REQUEST]: () => null,
    [SEND_CONSULTATION_RESPONSE]: (
      _,
      action: SendConsultationSuccessfulResponse
    ) => action.payload ?? null,
    [SEND_CONSULTATION_FAILURE]: () => null,
    [START_TRIPETTO]: () => null,
    [ORDER_ACTIONS.CLEANUP_ORDER_FLOW_STATE]: () => null,
  },
  null
);

const isConsultationValidForOrder = handleActions(
  {
    [SET_IS_CONSULTATION_VALID_FOR_ORDER]: (
      state: boolean,
      action: SetIsConsultationValidForOrderAction
    ) => action.payload,
    [START_TRIPETTO]: () => false,
    [ORDER_ACTIONS.CLEANUP_ORDER_FLOW_STATE]: () => false,
  },
  false
);

const consultation = combineReducers({
  responses: consultationResponses,
  token: consultationToken,
  isValidForOrder: isConsultationValidForOrder,
});

const orderNumber = handleActions(
  {
    [ON_ORDER_RESPONSE]: (_state: string, action: OrderResponseAction) =>
      action?.payload?.orderNumber,
    [FETCH_BRAINTREE_TOKEN_REQUEST]: () => "",
    [ON_ORDER]: () => "",
  },
  ""
);

const reducer: Reducer<State, any> = combineReducers({
  selectedOnlineService,
  onlineServices,
  selectedPack,
  selectedProduct,
  products,
  productToken,
  productTokenValid,
  fetching,
  braintreeToken,
  successfulOrder,
  orderErrorMessage,
  consultation,
  orderNumber,
});

export default reducer;
