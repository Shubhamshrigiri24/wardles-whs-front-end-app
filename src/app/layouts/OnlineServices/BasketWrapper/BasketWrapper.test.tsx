import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import BasketWrapper, { BasketWrapperStateProps } from "./BasketWrapper";
import { makeProduct } from "app/store/reducer/mock";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { Ed } from "../../../store/reducer/online/types";

const history = createMemoryHistory();

describe("<BasketWrapper />", () => {
  it("should match the snapshot", () => {
    const props = makeProps();
    const { container } = render(
      <Router history={history}>
        <BasketWrapper {...props} />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});

function makeProps(): BasketWrapperStateProps {
  const product = makeProduct();
  return {
    selectedOnlineService: Ed,
    basketItem: product.packs[1],
    product,
    children: [<h1 key={"1"}>Hello</h1>] as any,
  };
}
