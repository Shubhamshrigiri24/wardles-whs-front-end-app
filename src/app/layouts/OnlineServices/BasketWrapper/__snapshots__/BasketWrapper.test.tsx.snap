// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketWrapper /> should match the snapshot 1`] = `
<div>
  <div
    class="makeStyles-header-2"
  >
    <div
      class="makeStyles-logoContainer-6"
    >
      <img
        alt="WellLogo"
        class="makeStyles-wellLogo-7"
        src="Well.svg"
      />
      <div
        class="makeStyles-textContainer-8"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Pharmacy Services
        </p>
      </div>
    </div>
  </div>
  <hr
    class="MuiDivider-root makeStyles-divider-4"
  />
  <div>
    <div
      class="MuiGrid-root makeStyles-container-14 MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root makeStyles-mainPanelBreaksDownwards-16 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
      >
        <div
          class="MuiContainer-root makeStyles-container-1 MuiContainer-disableGutters MuiContainer-maxWidthMd"
        >
          <h1>
            Hello
          </h1>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-secondaryPanelBreaksUpwards-15 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          class="makeStyles-container-17"
        >
          <div
            class="MuiGrid-root makeStyles-paddingBottom-19 MuiGrid-container"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              <b>
                Your basket
              </b>
            </p>
            <div
              class="MuiGrid-root makeStyles-padding-20 MuiGrid-item MuiGrid-grid-xs-12"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1"
              >
                name-default 
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1"
              >
                label-default-pack-2
              </h6>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1"
              >
                Total
              </h6>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignRight"
              >
                Â£
                1.00
              </h6>
            </div>
          </div>
          <div
            class="makeStyles-root-22 makeStyles-infoType-23"
          >
            <div
              class="makeStyles-icon-24"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
            </div>
            <div
              class="makeStyles-message-25"
            >
              Includes ongoing support from our pharmacists.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
